pkg/publicapi/endpoint/requester/endpoints_cancel.go

```diff
@@ -30,22 +30,22 @@ import (
 //	@Failure				403				{object}	string
 //	@Failure				500				{object}	string
 //	@Router					/api/v1/requester/cancel [post]
-func (s *Endpoint) cancel(res http.ResponseWriter, req *http.Request) {
-	ctx := req.Context()
-	jobCancelPayload, err := signatures.UnmarshalSigned[model.JobCancelPayload](ctx, req.Body)
+func (s *Endpoint) cancel(c echo.Context) error {
+	ctx := c.Request().Context()
+	jobCancelPayload, err := signatures.UnmarshalSigned[model.JobCancelPayload](ctx, c.Request().Body)
 	if err != nil {
-		publicapi.HTTPError(ctx, res, err, http.StatusBadRequest)
-		return
+		publicapi.HTTPError(c, err, http.StatusBadRequest)
+		return nil
```

Comment:
Why do we return errors in someplaces and nil in other places? `publicapi.HTTPError` seems like a bad pattern.

Reply:
yeah it is because of `publicapi.HTTPError` implementation and how errors of type `bacerrors.ErrorResponse` are handled. Originally I only returned errors, and injected an error handler that handles `bacerrors.ErrorResponse` differently, but the change grew too much and was afraid I might've broken more stuff.

These are legacy APIs that I don't want to spend more time fixing or improving them